server/sonar-web/src/main/webapp/WEB-INF/lib/need_authentication.rb

```diff
@@ -153,13 +153,15 @@ def synchronize(username, password, details)
 
       # A user that is synchronized with an external system is always set to 'active' (see SONAR-3258 for the deactivation concept)
       user.active=true
-      # Note that validation disabled
+
+      # This save is to used to set a lock on the users table
       user.save(false)
 
       synchronize_groups(user)
-      user.notify_creation_handlers
-      user
+      user.save(false)
     end
+    user.notify_creation_handlers
```

Comment:
The point of this patch is that ```user.notify_creation_handlers``` is outside the RoR transaction. A comment should be added, with a link to JIRA ticket.